src/gilded_rose.js

```diff
@@ -13,50 +13,94 @@ items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 80));
-      if (items[i].quality > 0) {
-        if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-          items[i].quality = items[i].quality - 1
-        }
-      }
-    } else {
-      if (items[i].quality < 50) {
-        items[i].quality = items[i].quality + 1
-        if (items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].sell_in < 11) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-          if (items[i].sell_in < 6) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-        }
-      }
-    }
-    if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-      items[i].sell_in = items[i].sell_in - 1;
-    }
-    if (items[i].sell_in < 0) {
-      if (items[i].name != 'Aged Brie') {
-        if (items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].quality > 0) {
-            if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-              items[i].quality = items[i].quality - 1
-            }
-          }
-        } else {
-          items[i].quality = items[i].quality - items[i].quality
+  // for (var i = 0; i < items.length; i++) {
+  //   if (items[i].name != 'Aged Brie' && items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
+  //     if (items[i].quality > 0) {
+  //       if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
+  //         items[i].quality = items[i].quality - 1
+  //       }
+  //     }
+  //   } else {
+  //     if (items[i].quality < 50) {
+  //       items[i].quality = items[i].quality + 1
+  //       if (items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
+  //         if (items[i].sell_in <= 11) { // Made change from <  to <=
+  //           if (items[i].quality < 50) {
+  //             items[i].quality = items[i].quality + 1
+  //           }
+  //         }
+  //         if (items[i].sell_in <= 6) { // Made change from <  to <=
+  //           if (items[i].quality < 50) {
+  //             items[i].quality = items[i].quality + 1
+  //           }
+  //         }
+  //       }
+  //     }
+  //   }
+  //   if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
+  //     items[i].sell_in = items[i].sell_in - 1;
+  //   }
+  //   if (items[i].sell_in < 0) {
+  //     if (items[i].name != 'Aged Brie') {
+  //       if (items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
+  //         if (items[i].quality > 0) {
+  //           if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
+  //             items[i].quality = items[i].quality - 1
+  //           }
+  //         }
+  //       } else {
+  //         items[i].quality = items[i].quality - items[i].quality
+  //       }
+  //     } else {
+  //       if (items[i].quality < 50) {
+  //         items[i].quality = items[i].quality + 1
+  //       }
+  //     }
+  //   }
+  // }
+var numUpdateby = 1
```

Comment:
Use `let` not `var`
The scope of `numUpdateby` can be narrowed. Move it inside the `forEach` callback.